### PR TITLE
Assure datadir exists at start

### PR DESCRIPTION
**Pull request by LefterisJP**
I was getting a segfault at the writting of the config file if there was
no datadir (~/.ethereum) present. Before anything we need to make sure
that this directory exists.

### CODE REVIEW

File: eth/main.cpp [hunk 1]
```diff
@@ -340,6 +341,10 @@ int main(int argc, char** argv)
 	/// Wallet password stuff
 	string masterPassword;
 
+	// make sure data dir exists
+	if (!boost::filesystem::create_directory(getDataDir()))
+		BOOST_THROW_EXCEPTION(FileError());
```
Comment (subtly) [inline eth/main.cpp]: Does create_directory throw?
As this is CLI, would rather see an error message output to the user with exit() rather than a crash.

Comment (LefterisJP) [inline eth/main.cpp]: We already throw a lot of those uncaught exceptions around in the CLI from inside other functions that are called in main(). But yes I agree. A message is in order

**Comment by gavofyork** [comment]
this is the wrong place for it.

State & Blockchain definitely create paths they need. The likely candidate is KeyManager, but to be sure, get a backtrace from the exception and figure out what code is trying to open a file in a non-existant directory.

**Comment by LefterisJP** [comment]
Yes it was the key manager. Specifically here:

```
	KeyManager keyManager;
	for (auto const& s: passwordsToNote)
		keyManager.notePassword(s);

	{
		RLPStream config(2);
		config << signingKey << beneficiary;
		writeFile(configFile, config.out());
	}
```

That `writeFile()` fails. Do you want me to move it before that? Or put it in the KeyManager class?

**Comment by gavofyork** [comment]
yes.

**Comment by gavofyork** [comment]
fixed with #2261.

**Pull request closed without merging**